docs/dev/cluster-api.md

```diff
@@ -33,3 +34,96 @@ To build an `openshift-install` binary with Cluster API bundled:
 - Set `export OPENSHIFT_INSTALL_CLUSTER_API=y`
     - Optionally `export SKIP_TERRAFORM=y` if you don't need to use Terraform.
 - Run `./hack/build.sh`, the binary is then produced in `bin/openshift-install`.
+
+### Platform Onboarding
+
+This is a general guide for onboarding a new platform to provision infrastructure with cluster api.
+
+#### CAPI Cloud Provider Controller Dependencies
+
+Providers are vendored to [`cluster-api/providers/<name>`][providerDir]. The README contains
+detailed guidelines on how to vendor a new platform.
+
+#### CAPI Cloud Provider CRDs
+
+To add the CRDs for the CAPI cloud provider to the local control plane, add them to the
+[`/data/data/cluster-api`][crdDir] dir. Typically these CRDs are published as an
+`infrastructure-components.yaml` file as part of the release of the CAPI controller. The
+yaml files in this directory will automatically be added to the Installer's local
+control plane.
+
+#### Run the CAPI Controller
+
+In order for the Installer to run the controller, which was added in a previous step, add
+the details for the controller in [this switch case in pkg/clusterapi/system.go][runController].
+The controller will be unpacked and run based on the provider name; flags and environment variables
+can be specified.
```

Comment:
nit: Consider adding a note about finding these flags in each provider Controller